src/main/io/osd.c

```diff
@@ -0,0 +1,723 @@
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+
+#include "platform.h"
+#include "scheduler/scheduler.h"
+
+#include "common/axis.h"
+#include "common/color.h"
+#include "common/atomic.h"
+#include "common/maths.h"
+#include "common/typeconversion.h"
+
+#include "drivers/nvic.h"
+
+#include "drivers/sensor.h"
+#include "drivers/system.h"
+#include "drivers/gpio.h"
+#include "drivers/light_led.h"
+#include "drivers/sound_beeper.h"
+#include "drivers/timer.h"
+#include "drivers/serial.h"
+#include "drivers/serial_softserial.h"
+#include "drivers/serial_uart.h"
+#include "drivers/accgyro.h"
+#include "drivers/compass.h"
+#include "drivers/pwm_mapping.h"
+#include "drivers/pwm_rx.h"
+#include "drivers/adc.h"
+#include "drivers/bus_i2c.h"
+#include "drivers/bus_spi.h"
+#include "drivers/inverter.h"
+#include "drivers/flash_m25p16.h"
+#include "drivers/sonar_hcsr04.h"
+#include "drivers/gyro_sync.h"
+#include "drivers/usb_io.h"
+#include "drivers/transponder_ir.h"
+#include "drivers/sdcard.h"
+
+#include "rx/rx.h"
+
+#include "io/beeper.h"
+#include "io/serial.h"
+#include "io/flashfs.h"
+#include "io/gps.h"
+#include "io/escservo.h"
+#include "io/rc_controls.h"
+#include "io/gimbal.h"
+#include "io/ledstrip.h"
+#include "io/display.h"
+#include "io/asyncfatfs/asyncfatfs.h"
+#include "io/transponder_ir.h"
+#include "io/osd.h"
+
+#include "sensors/sensors.h"
+#include "sensors/sonar.h"
+#include "sensors/barometer.h"
+#include "sensors/compass.h"
+#include "sensors/acceleration.h"
+#include "sensors/gyro.h"
+#include "sensors/battery.h"
+#include "sensors/boardalignment.h"
+#include "sensors/initialisation.h"
+
+#include "telemetry/telemetry.h"
+#include "blackbox/blackbox.h"
+
+#include "flight/pid.h"
+#include "flight/imu.h"
+#include "flight/mixer.h"
+#include "flight/failsafe.h"
+#include "flight/navigation.h"
+
+#include "config/runtime_config.h"
+#include "config/config.h"
+#include "config/config_profile.h"
+#include "config/config_master.h"
+
+#ifdef USE_HARDWARE_REVISION_DETECTION
+#include "hardware_revision.h"
+#endif
+
+#include "build_config.h"
+#include "debug.h"
+
+#ifdef OSD
+
+#include "drivers/max7456.h"
+#include "drivers/max7456_symbols.h"
+
+#ifdef USE_RTC6705
+#include "drivers/vtx_soft_spi_rtc6705.h"
+#endif
+
+#include "common/printf.h"
+
+#define MICROSECONDS_IN_A_SECOND (1000 * 1000)
+
+#define OSD_UPDATE_FREQUENCY (MICROSECONDS_IN_A_SECOND / 5)
+#define OSD_LINE_LENGTH 30
+
+#define STICKMIN 10
+#define STICKMAX 90
+
+
+static uint32_t next_osd_update_at = 0;
+static uint32_t armed_seconds = 0;
+static uint32_t armed_at = 0;
+static bool armed = false;
+
+static uint8_t current_page = 0;
+static uint8_t sticks[] = {0,0,0,0};
+static char string_buffer[30];
+static uint8_t cursor_row = 255;
+static uint8_t cursor_col = 0;
+static bool in_menu = false;
+static bool activating_menu = false;
+extern uint16_t rssi;
+
+
+#ifdef USE_RTC6705
+void update_vtx_band(bool increase, uint8_t col) {
```

Comment:
`bool` parameters make for unmaintainable code. If a developer sees `update_vtx_band(true, ...)` there is a WTF moment. Instead define `enum`s , something like `VTX_BAND_INCREASE` and `VTX_BAND_DECREASE`, so the code would then read `update_vtx_band(VTX_BAND_INCREASE...)` - the developer would then know what was going on.